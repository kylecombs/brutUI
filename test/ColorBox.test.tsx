import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { ColorBox, colors } from "../src";

afterEach(cleanup);

describe("ColorBox", () => {
  it("Renders ColorBox correctly", () => {
    const { asFragment } = render(
      <>
        <ColorBox key={colors[0].bgClass} color={colors[0]} />
        <ColorBox key={colors[1].bgClass} color={colors[1]} />
        <ColorBox key={colors[2].bgClass} color={colors[2]} />
        <ColorBox key={colors[3].bgClass} color={colors[3]} />
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
