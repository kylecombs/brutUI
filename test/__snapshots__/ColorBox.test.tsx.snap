// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox Renders ColorBox correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white"
  >
    <div
      class="h-20 bg-gray-25 rounded-t-lg"
    />
    <div
      class="mx-3 my-3"
    >
      <p
        class="text-lg font-medium text-gray-900"
      >
        25
      </p>
      <p
        class="text-md font-normal text-gray-500"
      >
        #FCFCFD
      </p>
    </div>
  </div>
  <div
    class="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white"
  >
    <div
      class="h-20 bg-gray-50 rounded-t-lg"
    />
    <div
      class="mx-3 my-3"
    >
      <p
        class="text-lg font-medium text-gray-900"
      >
        50
      </p>
      <p
        class="text-md font-normal text-gray-500"
      >
        #F9FAFB
      </p>
    </div>
  </div>
  <div
    class="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white"
  >
    <div
      class="h-20 bg-gray-100 rounded-t-lg"
    />
    <div
      class="mx-3 my-3"
    >
      <p
        class="text-lg font-medium text-gray-900"
      >
        100
      </p>
      <p
        class="text-md font-normal text-gray-500"
      >
        #F2F4F7
      </p>
    </div>
  </div>
  <div
    class="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white"
  >
    <div
      class="h-20 bg-gray-200 rounded-t-lg"
    />
    <div
      class="mx-3 my-3"
    >
      <p
        class="text-lg font-medium text-gray-900"
      >
        200
      </p>
      <p
        class="text-md font-normal text-gray-500"
      >
        #E4E7EC
      </p>
    </div>
  </div>
</DocumentFragment>
`;
